lib/asset/transferpart.js

```diff
@@ -34,13 +35,13 @@ class TransferPart {
      */
     constructor(transferAsset, targetUrls, contentRange, targetHeaders) {
         if (!(transferAsset instanceof TransferAsset)) {
-            throw Error(`Invalid transfer asset: ${transferAsset}`);
+            throw new IllegalArgumentError("Invalid transfer asset", transferAsset);
         }
         if (!Array.isArray(targetUrls) || (targetUrls.length === 0)) {
-            throw Error(`Invalid target urls: ${targetUrls}`);
+            throw new IllegalArgumentError("Invalid target urls", targetUrls);
```

Comment:
For this one it would be good to elaborate the error text - because from the text you can't know if the issue is that the URLs are not valid (expired?) or that there were no URLs provided.